kyuubi-common/src/main/scala/org/apache/kyuubi/service/AbstractBackendService.scala

```diff
@@ -160,7 +160,18 @@ abstract class AbstractBackendService(name: String)
     val operation = sessionManager.operationManager.getOperation(operationHandle)
     if (operation.shouldRunAsync) {
       try {
-        operation.getBackgroundHandle.get(timeout, TimeUnit.MILLISECONDS)
```

Comment:
the fix here is wrong, because operation.getBackgroundHandle is null forever

```diff
+++ b/externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/operation/ExecuteStatement.scala
@@ -30,7 +30,7 @@ class ExecuteStatement(
     session: Session,
     override val statement: String,
     confOverlay: Map[String, String],
-    override val shouldRunAsync: Boolean,
+    runAsync: Boolean,
     queryTimeout: Long)
   extends HiveOperation(OperationType.EXECUTE_STATEMENT, session) {

@@ -39,9 +39,11 @@ class ExecuteStatement(
       hive,
       statement,
       confOverlay.asJava,
-      shouldRunAsync,
+      runAsync,
       queryTimeout)
   }

   EventBus.post(HiveOperationEvent(this))
+
+  override def shouldRunAsync: Boolean = runAsync && getBackgroundHandle != null
 }
```